src/averageAges.js

```diff
@@ -1,61 +1,84 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
+
+function calculateAge(person) {
+  return person.died - person.born;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredAges = people.filter((person) =>
+    person.sex === 'm'
+    && (century ? Math.ceil(person.died / 100) === century : true)
+  );
+
+  return filteredAges.reduce((sum, person) =>
+    sum + calculateAge(person), 0) / filteredAges.length;
 }
 
-/**
+/*
  * Implement calculateWomenAverageAge function
  *
  * Function returns average age of women in array. If `withChildren` is
  * specified then function calculates average age only for women with children
  *
  * Hint: To check if a woman has children you should find someone who mention
  * her as mother.
- *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const findWomen = people.filter(item => item.sex === 'f');
+  const motherNames = people.map(item => item.mother);
```

Comment:
Consider renaming 'item' to 'person' for better readability and consistency with the rest of the code.